src/ucp/wireup/wireup.c

```diff
@@ -973,11 +973,20 @@ ucp_wireup_replay_pending_request(uct_pending_req_t *self, ucp_ep_h ucp_ep)
                       req->send.proto_config->proto->name);
         ucp_proto_request_restart(req);
     } else {
-        ucp_trace_req(req, "replay proto %s lane %d",
+        ucp_trace_req(req, "replay proto on %s lane %d by %s",
                       (req->flags & UCP_REQUEST_FLAG_PROTO_SEND) ?
                       req->send.proto_config->proto->name : "N/A",
-                      req->send.lane);
-        ucp_request_send(req);
+                      req->send.lane,
+                      ucs_debug_get_symbol_name(req->send.uct.func));
+        if (req->send.uct.func == ucp_am_contig_short) {
+            ucs_assert(!(req->flags & UCP_REQUEST_FLAG_PROTO_SEND));
+            ucp_am_short_request_replay(req);
+        } else if (req->send.uct.func == ucp_am_contig_short_reply) {
+            ucs_assert(!(req->flags & UCP_REQUEST_FLAG_PROTO_SEND));
+            ucp_am_short_reply_request_replay(req);
+        } else {
+            ucp_request_send(req);
```

Comment:
why is this relevant for short sends only with AM API?

Reply:
technically the issue is relevant for any protocol, AM/short is real use case, proto v2 solves the issue by design